### PR TITLE
Documentation for .NET Lambda Instrumentation

### PR DESCRIPTION
**Pull request by nhinsch**
### What does this PR do?
- Update docs to include support for .NET Lambda functions
- Fix/update a few other things throughout the Lambda docs

### Preview
- https://docs-staging.datadoghq.com/ngh/dot-net/serverless/distributed_tracing
- https://docs-staging.datadoghq.com/ngh/dot-net/serverless/enhanced_lambda_metrics
- https://docs-staging.datadoghq.com/ngh/dot-net/serverless/guide/extension_motivation
- https://docs-staging.datadoghq.com/ngh/dot-net/serverless/installation/dotnet
- https://docs-staging.datadoghq.com/ngh/dot-net/serverless/installation/nodejs
- https://docs-staging.datadoghq.com/ngh/dot-net/serverless/installation/python
- https://docs-staging.datadoghq.com/ngh/dot-net/serverless/libraries_integrations/extension
---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/serverless/installation/dotnet.md [hunk 1]
```diff
@@ -87,11 +206,27 @@ LambdaLogger.Log(JsonConvert.SerializeObject(myMetric));
...
+#### Tagging
+
+Although it's optional, Datadog recommends tagging your serverless applications with the `env`, `service`, and `version` tags following the [unified service tagging documentation][8].
+
+## Collect logs from AWS serverless resources
 
+Serverless logs generated by managed resources besides AWS Lambda functions can be hugely valuable in helping identify the root cause of issues in your serverless applications. Datadog recommends you forward logs from the following managed resources in your environment:
+- APIs: API Gateway, AppSync, ALB.
+- Queues & Streams: SQS, SNS, Kinesis.
+- Data Stores: DynamoDB, S3, RDS, etc.
```
Comment (cswatt) [inline content/en/serverless/installation/dotnet.md:218]:
```suggestion
- Data stores: DynamoDB, S3, RDS, etc.
```

File: content/en/serverless/libraries_integrations/library.md [hunk 2]
```diff
@@ -32,8 +32,8 @@ The Datadog Lambda Library is responsible for:
 - Submitting [custom metrics][2] (synchronously and asynchronously).
 - Enabling [Datadog APM and Distributed Tracing][3] for Node.js, Python, and Ruby.
 
-If you are using the Datadog Lambda library for **Ruby** or **Java**, you **must** also install and configure the Datadog Forwarder to ingest traces, enhanced Lambda metrics, or custom metrics (asynchronously) from your Lambda functions.
-If you are using the Datadog Lambda library for **Python**, **Node**, or **Go**, you may use the [Datadog Lambda Extension][4] to ingest traces, enhanced Lambda metrics, or custom metrics; or you may continue to use the Datadog Forwarder.
+If you are using the Datadog Lambda library for **Ruby**, you **must** also install and configure the Datadog Forwarder to ingest traces, enhanced Lambda metrics, or custom metrics (asynchronously) from your Lambda functions.
+If you are using the Datadog Lambda library for **Python**, **Node**, **Go**, or **Java** you may use the [Datadog Lambda Extension][4] to ingest traces, enhanced Lambda metrics, or custom metrics; or you may continue to use the Datadog Forwarder.
```
Comment (cswatt) [inline content/en/serverless/libraries_integrations/library.md:36]:
```suggestion
If you are using the Datadog Lambda library for **Python**, **Node**, **Go**, or **Java**, you may use the [Datadog Lambda Extension][4] to ingest traces, enhanced Lambda metrics, or custom metrics. Alternatively, you may continue to use the Datadog Forwarder.
```

File: content/en/serverless/distributed_tracing/_index.md [hunk 3]
```diff
@@ -100,11 +100,13 @@ For Java serverless applications, Datadog recommends [installing Datadog's traci
...
-Datadog recommends you configure tracing for your .NET AWS Lambda serverless applications by [installing our AWS X-Ray integration][3].
+The tracing library for .NET supports:
+- Tracing HTTP requests invoking downstream Lambda functions or containers.
+- Tracing dozens of additional out-of-the-box [.NET][18] libraries.
 
-Learn more about tracing through .NET Azure serverless applications [here][16].
+For .NET serverless applications, Datadog recommends [installing Datadog's tracing libraries][7]. If your application requires AWS X-Ray active tracing in AWS managed services such as API Gateway or Step Functions, you may want to consider instead using [Datadog APM with AWS X-Ray tracing][3].
 
-*Looking to trace through .NET AWS Lambda serverless applications with Datadog tracing libraries? Open a feature request [here][9].*
+Learn more about tracing through .NET Azure serverless applications [here][16].
```
Comment (cswatt) [inline content/en/serverless/distributed_tracing/_index.md:109]:
```suggestion
Learn more about [tracing through .NET Azure serverless applications][16].
```

**Review by cswatt — CHANGES_REQUESTED** [review]

File: content/en/serverless/installation/dotnet.md [hunk 4]
```diff
@@ -13,69 +13,188 @@ further_reading:
   text: 'Submitting Custom Metrics from Serverless Applications'
 ---
 
-## Prerequisites
+To instrument your .NET serverless application, you must use **either** the [Datadog Lambda Extension][1] (beta) or the [Datadog Forwarder Lambda function][2].
 
-To instrument your .NET serverless application, you must install **either** the [Datadog Forwarder Lambda function][1] or (beta) the [Datadog Lambda Extension][2].
+If you use the Datadog Lambda Extension, you can use Datadog's native .NET APM tracer to instrument your function. If you use the Datadog Forwarder Lambda function, you can use Datadog's integration with AWS X-Ray tracing.
```
Comment (Hesperide) [inline content/en/serverless/installation/dotnet.md:18]: Should we make this simpler by saying if you want to use tracing, you need to use the Extension?

File: content/en/serverless/installation/dotnet.md [hunk 5]
```diff
@@ -13,69 +13,188 @@ further_reading:
...
+        CORECLR_ENABLE_PROFILING: 1
+        CORECLR_PROFILER: "{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"
+        CORECLR_PROFILER_PATH: "/opt/datadog/Datadog.Trace.ClrProfiler.Native.so"
+        DD_DOTNET_TRACER_HOME: "/opt/datadog"
+    ```
 
+    Replace `<YOUR_DD_API_KEY>` with your Datadog API key, found on the [API Management page][1].
 
-[1]: /serverless/libraries_integrations/extension/
+2. Optionally add `service` and `env` tags with appropriate values to your function.
```
Comment (Hesperide) [inline content/en/serverless/installation/dotnet.md:52]: Should we add something after this to make it clear that the reader is done and doesn't need to follow the Forwarder instructions?

File: content/en/serverless/installation/dotnet.md (hunk 4 above)
Comment (nhinsch) [reply content/en/serverless/installation/dotnet.md:18]: I'll leave it to you. However, some customers might want to keep using the Forwarder (e.g. they are using ARM or they don't want to use a beta product) and these customers probably do want to use the X-Ray integration.

File: content/en/serverless/installation/dotnet.md (hunk 5 above)
Comment (nhinsch) [reply content/en/serverless/installation/dotnet.md:52]: I added a line at the beginning of the Forwarder section to this effect, do you think it's not enough?

**Pull request merged**